nsfc-plugin/src/main/java/org/osc/controller/nsfc/utils/OsCalls.java

```diff
@@ -0,0 +1,197 @@
+
+   public FlowClassifier updateFlowClassifier(String flowClassifierId, FlowClassifier flowClassifier) {
+       checkArgument(flowClassifierId != null, "null passed for %s !", "Flow Classifier Id");
+       checkArgument(flowClassifier != null, "null passed for %s !", "Flow Classifier");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       flowClassifier = flowClassifier.toBuilder().id(null).projectId(null).build();
+
+       return this.osClient.sfc().flowclassifiers().update(flowClassifierId, flowClassifier);
+   }
+
+   public PortChain updatePortChain(String portChainId, PortChain portChain) {
+       checkArgument(portChainId != null, "null passed for %s !", "Port Chain Id");
+       checkArgument(portChain != null, "null passed for %s !", "Port Chain");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       portChain = portChain.toBuilder().id(null).projectId(null).chainParameters(null).chainId(null).build();
+
+       portChain = this.osClient.sfc().portchains().update(portChainId, portChain);
+       return fixPortChainCollections(portChain);
+   }
+
+   public PortPairGroup updatePortPairGroup(String portPairGroupId, PortPairGroup portPairGroup) {
+       checkArgument(portPairGroupId != null, "null passed for %s !", "Port Pair Group Id");
+       checkArgument(portPairGroup != null, "null passed for %s !", "Port Pair Group");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       portPairGroup  = portPairGroup.toBuilder().id(null).projectId(null).portPairGroupParameters(null).build();
+
+       portPairGroup = this.osClient.sfc().portpairgroups().update(portPairGroupId, portPairGroup);
+       return portPairGroup;
+   }
+
+   public PortPair updatePortPair(String portPairId, PortPair portPair) {
+       checkArgument(portPairId != null, "null passed for %s !", "Port Pair Id");
+       checkArgument(portPair != null, "null passed for %s !", "Port Pair");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       portPair = portPair.toBuilder().id(null).projectId(null).build();
+       portPair = this.osClient.sfc().portpairs().update(portPairId, portPair);
+       return portPair;
+   }
+
+   public Port updatePort(Port port) {
+       return this.osClient.networking().port().update(port);
+   }
+
+   public ActionResponse deleteFlowClassifier(String flowClassifierId) {
+       return this.osClient.sfc().flowclassifiers().delete(flowClassifierId);
+   }
+
+   public ActionResponse deletePortChain(String portChainId) {
+       ActionResponse response = this.osClient.sfc().portchains().delete(portChainId);
+       return response;
+   }
+
+   public ActionResponse deletePortPairGroup(String portPairGroupId) {
+       ActionResponse response = this.osClient.sfc().portpairgroups().delete(portPairGroupId);
+       return response;
+   }
+
+   public ActionResponse deletePortPair(String portPairId) {
```

Comment:
we could encapsute the action response and just throw exception right?

i would use the delete method to handle 404 error(ignore) and wrap and throw exceptions for other cases, instead of handling this at the caller...